src/main/scala/mesosphere/marathon/core/task/update/impl/TaskStatusUpdateProcessorImpl.scala

```diff
@@ -43,13 +47,19 @@ class TaskStatusUpdateProcessorImpl @Inject() (
 
     val now = clock.now()
     val taskId = Task.Id(status.getTaskId)
+    val marathonTaskStatus = MarathonTaskStatus(status)
 
     taskTracker.task(taskId).flatMap {
       case Some(task) =>
         val taskStateOp = TaskStateOp.MesosUpdate(task, status, now)
         stateOpProcessor.process(taskStateOp).flatMap(_ => acknowledge(status))
 
-      case None if killWhenUnknown(status) =>
+      case None if terminalUnknown(marathonTaskStatus) =>
+        log.warn("Received terminal status update for unknown {}", taskId)
+        eventStream.publish(UnknownTaskTerminated(taskId, taskId.runSpecId, marathonTaskStatus))
+        acknowledge(status)
+
+      case None if killWhenUnknown(marathonTaskStatus) =>
```

Comment:
Which wouldn't be 100% correct – `Unreachable` is a non-terminal state, but killing a task that's unreachable will just lead to another status update that says it's `Unreachable`. There's a reason for being so explicit, but it's obviously not clear, so I'll document that.